protocol/kafka_confluent/v2/protocol.go

```diff
@@ -128,20 +133,13 @@ func (p *Protocol) Send(ctx context.Context, in binding.Message, transformers ..
 		kafkaMsg.Key = []byte(messageKey)
 	}
 
-	err = WriteProducerMessage(ctx, in, kafkaMsg, transformers...)
-	if err != nil {
-		return err
+	if err = WriteProducerMessage(ctx, in, kafkaMsg, transformers...); err != nil {
+		return fmt.Errorf("create producer message: %w", err)
 	}
 
-	err = p.producer.Produce(kafkaMsg, p.producerDeliveryChan)
-	if err != nil {
+	if err = p.producer.Produce(kafkaMsg, nil); err != nil {
 		return err
```

Comment:
I wanted to return a `kafka.Error` for fear that users might also want to check its type using an assert statement like [this](https://github.com/confluentinc/confluent-kafka-go/blob/master/examples/producer_example/producer_example.go#L90-L98).

Reply:
That's why I'm wrapping the error with `%w` to allow Go idiomatic use of https://pkg.go.dev/errors#As

Reply:
@yanmxa WDYT?

Reply:
While the type of [`kafka.Error`](https://github.com/confluentinc/confluent-kafka-go/blob/cb2701e8b6f43be0cefb22da5c56c6856f7f0899/kafka/error.go#L35-L41) is not the type of error we typically refer to. 
It's an object that only implements the [`error` interface](https://github.com/confluentinc/confluent-kafka-go/blob/master/kafka/error.go#L88-L92). So once the `kafka.Error` is formatted with `%w`, then it can't be unwrapped back to `kafka.Error`, like: `interface conversion: error is *fmt.wrapError, not kafka.Error`

Reply:
Perhaps we're talking past each other :)

This doesn't work?

```go
                 var kafkaError kafka.Error
                 if errors.As(err, &kafkaError) {
			// use kafkaError to handle and access fields
		}
```

Reply:
@yanmxa ICYMI ^